madara/crates/client/block_import/src/verify_apply.rs

```diff
@@ -325,6 +328,52 @@ fn block_hash(
     Ok((block_hash, header))
 }
 
+/// Reorgs the blockchain from its current tip back to `new_tip` from which a new fork can then be
+/// played on top.
+///
+/// Returns the result of the reorg, which describes the part of the chain that was orphaned.
+#[allow(dead_code)] // not used yet outside of tests
+fn reorg(
```

Comment:
Thoughts on calling this `reorg()`?

It's really just reverting. Other than some test cases, there is no actual reorg going on.

It might help to have some clear distinction between the two, so I'll define it here (let me know if you disagree):

* `Revert`: Set the current tip of the blockchain to some direct ancestor, dropping all blocks in between
* `Reorg`: Set the current tip of the blockchain to a block on another fork of the blockchain. Achieved by first reverting to the most recent common ancestor, then applying new blocks until the desired one is reached.

With this distinction in mind, I think this should be called revert and that this current PR isn't so much about reorgs (again, other than the tests).

Reply:
should go into the backend imo, and be called revert_to